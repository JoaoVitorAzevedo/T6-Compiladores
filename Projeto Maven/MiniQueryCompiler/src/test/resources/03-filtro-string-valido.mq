LOAD "data.json"
FILTER .nome == "Maria"
SELECT .email
SAVE AS "email_da_maria.json"