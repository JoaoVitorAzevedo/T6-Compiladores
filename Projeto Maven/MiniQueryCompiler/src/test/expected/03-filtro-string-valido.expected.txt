ANÁLISE SEMÂNTICA VÁLIDA

CÓDIGO JS GERADO:
// Código gerado automaticamente
const fs = require('fs');

const input = JSON.parse(fs.readFileSync('data.json', 'utf-8'));
const filteredData = input.filter(item => item.nome === "Maria");
const result = filteredData.map(item => ({
  email: item.email
}));
fs.writeFileSync('email_da_maria.json', JSON.stringify(result, null, 2));